src/Shared.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Core.c
+ *
+ *  Created on: 9/09/2021
+ *      Author: James Brown
+ */
+
+#include "shared/Shared.h"
+#include "shared/Error.h"
+
+#include "MCU_Init.h"
+
+HAL_StatusTypeDef EXT(CORE_getTime)(RTC_TimeTypeDef *time, RTC_DateTypeDef *date, uint32_t format) {
```

Comment:
This is very different from EnduroSat's implementation: [Link to Koios OBC since this isnt implemented in Kessler's version](https://github.com/APSS-Koios/apss3-obc/blob/4c764ae84bb48462e030a7fe35faf1f08d98dfd8/Middlewares/ESTTC.c#L1124)

Reply:
Yeah I did see this. @Evander7 also had a similar implementation. The synchro appears to only be needed if the time had recently been updated or initialised. So I am not sure if it would be needed. The delay also seems unneeded, I can see no reference anywhere to a delay requirement in any document for the HAL interface; and I would assume that any delay requirements have already been met if needed by a synchro call:

From the HAL source:
> To read the calendar through the shadow registers after Calendar 
initialization, calendar update or after wake-up from low power modes 
the software must first clear the RSF flag. The software must then 
wait until it is set again before reading the calendar, which means 
that the calendar registers have been correctly copied into the 
RTC_TR and RTC_DR shadow registers.The HAL_RTC_WaitForSynchro() function 
implements the above software sequence (RSF clear and RSF check).

Additionally, `HAL_RTC_WaitForSynchro` requires write protection be dissabled on the RTC registers. Which does not appear to happen in the endurosat code at any point, so I'm not sure the call will have any effect anyway.

I would be interested in testing if this is the case. But for now I think leaving the code as it is is the best bet, as this seems to be the correct implementation as per the spec.